source/main/physics/BeamFactory.cpp

```diff
@@ -1023,18 +1008,14 @@ void ActorManager::UpdateActors(Actor* player_actor, float dt)
             return; // Skip UpdatePhysicsSimulation()
     }
 
-    if (m_sim_thread_pool)
-    {
-        auto func = std::function<void()>([this]()
-            {
-                this->UpdatePhysicsSimulation();
-            });
-        m_sim_task = m_sim_thread_pool->RunTask(func);
-    }
-    else
-    {
-        this->UpdatePhysicsSimulation();
-    }
+    auto func = std::function<void()>([this]()
+        {
+            this->UpdatePhysicsSimulation();
+        });
+    m_sim_task = gEnv->threadPool->RunTask(func);
+
+    if (!RoR::App::app_async_physics.GetActive())

```

Comment:
This reminds me of the "AsynchronousPhysics" setting in 0.4.0.7 which I removed as broken by design. Unless I'm missing something, it should cause a race condition under heavier physics workload. `join()`ing the sim. tasks here guards you against overlapping physics updates - without it, you may start another round of updates before the previous one finishes, eventually causing the same actor (nodes, beams, engine...) to be modified by 2 threads, the current step and the previous step. Correct me if I'm wrong.